### PR TITLE
Improval of the curl tool in the Offline Installation

### PR DESCRIPTION
**Pull request by davidcr01**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-packages/issues/2049|

## Description

This PR modifies all the `curl` commands that fetch big resources, mainly those included in the Offline Installation, using the `-I` parameter of `curl` to perform a `HEAD` request instead of a `GET` request.

This modification makes the Offline Installation to take less time, avoiding downloading the resources twice.

A Github Action has been added to test the Offline Installation. The action is in the `actions/offline-installation` folder.
The action has been debugged in https://github.com/davidcr01/wazuh-packages/actions.

### CODE REVIEW

File: .github/workflows/offline-installation.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel previous runs
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          cancel_others: 'true'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          skip_after_successful_duplicate: 'false'
+
+      - uses: actions/checkout@v2
```
Comment (verdx) [inline .github/workflows/offline-installation.yml:18]: Maintain a line between all steps for consistency

File: .github/workflows/offline-installation.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Build wazuh-install script and use pre-release packages
+        working-directory: ./unattended_installer
+        run: |
+          bash builder.sh -i -d
+          sed -i 's|wazuh_major="4\.5"|wazuh_major="4\.4"|g' wazuh-install.sh
+          sed -i 's|wazuh_version="4\.5\(.*\)"|wazuh_version="4\.4\1"|g' wazuh-install.sh
+        shell: bash
```
Comment (verdx) [inline .github/workflows/offline-installation.yml:25]: Is it necessary to specify the shell?

File: .github/workflows/offline-installation.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          name: script
+          path: |
+            unattended_installer/wazuh-install.sh
+          if-no-files-found: error
+          
+  Test-offline-installation-debian:
+    runs-on: ubuntu-latest
+    needs: Build-wazuh-install-script
+    steps:
+      - name: Cancel previous runs
```
Comment (verdx) [inline .github/workflows/offline-installation.yml:38]: I don't think it is necessary to call the step on every job, just once for each workflow

**Review by verdx — CHANGES_REQUESTED** [review]

**Review by DFolchA — APPROVED** [review]
LGTM

File: .github/workflows/offline-installation.yml (hunk 3 above)
Comment (davidcr01) [reply .github/workflows/offline-installation.yml]: True. Done in https://github.com/wazuh/wazuh-packages/pull/2071/commits/1eda46be9517179d44c28ae854e32b078d880347.

File: .github/workflows/offline-installation.yml (hunk 2 above)
Comment (davidcr01) [reply .github/workflows/offline-installation.yml]: The default shell is bash. Done in https://github.com/wazuh/wazuh-packages/pull/2071/commits/1eda46be9517179d44c28ae854e32b078d880347.

**Review by verdx — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]
LGTM

**Comment by alberpilot** [comment]
We will merge this PR when a new issue is created with the following request: re-use the off-line installation steps to create an official offline installation with `wazuh-install.sh`.

**Review by DFolchA — APPROVED** [review]

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**